### PR TITLE
key locker, to support holding locks by keys

### PR DESCRIPTION
**Pull request by OneOfOne**

### CODE REVIEW

File: keylocker.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package atoms
+
+import "sync"
+
+// KeyLocker allows to hold locks by key.
+type KeyLocker struct {
+	m   map[string]*sync.RWMutex
+	mux sync.RWMutex
```
Comment (itsmontoya) [inline keylocker.go]: When including a mux, it should be the very first declaration per our best practices

File: keylocker.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+// RWMutex returns a RWMutex for the specifc key.
+func (kl *KeyLocker) RWMutex(key string) (m *sync.RWMutex) {
+	kl.mux.RLock()
```
Comment (itsmontoya) [inline keylocker.go]: Can you move this into a kl.get(key) func? That way you can re-use it for delete and RwMutex

**Review by itsmontoya — CHANGES_REQUESTED** [review]

File: keylocker.go (hunk 2 above)
Comment (OneOfOne) [reply keylocker.go]: Can't reuse it for delete, the code is different :)

**Review by itsmontoya — APPROVED** [review]

**Comment by itsmontoya** [comment]
Boom :+1:  :100:

**Pull request merged**